### DISCUSSION TITLE
xApp help

### DISCUSSION THREAD

**Discussion by kenjo138** [srsran/srsRAN_Project#176] [General Help]
I was able to setup and run the O-RAN NearRT-RIC build. After running the core, gnB, FlexRIC and UE, I am having trouble running the xapp_kpm_moni xApp . I am not sure what I am missing. I run the ./examples/xApp/c/monitor/xapp_kpm_moni command from within the flexric directory, but I get a ./xapp_kpm_moni: command not found message. I also tried xapp_kpm_moni.c and along with gcc.

If I run sudo gcc ./xapp_kpm_moni.c I get the following message:

/usr/bin/ld: /tmp/ccGZO7Lz.o: in function `DF_60_line___impl.5555':
xapp_kpm_moni.c:(.text+0x157): undefined reference to `free_e2_node_arr'
/usr/bin/ld: /tmp/ccGZO7Lz.o: in function `main':
xapp_kpm_moni.c:(.text+0x1e9): undefined reference to `init_fr_args'
/usr/bin/ld: xapp_kpm_moni.c:(.text+0x1f8): undefined reference to `init_xapp_api'
/usr/bin/ld: xapp_kpm_moni.c:(.text+0x207): undefined reference to `e2_nodes_xapp_api'
/usr/bin/ld: xapp_kpm_moni.c:(.text+0x3d6): undefined reference to `report_sm_xapp_api'
/usr/bin/ld: xapp_kpm_moni.c:(.text+0x472): undefined reference to `rm_report_sm_xapp_api'
/usr/bin/ld: xapp_kpm_moni.c:(.text+0x4b6): undefined reference to `try_stop_xapp_api'
/usr/bin/ld: xapp_kpm_moni.c:(.text+0x4d8): undefined reference to `cleanup_deferred'
collect2: error: ld returned 1 exit status

I don't know if I am missing any libraries or if I am not properly setting this up. Any assistance would be appreciated.

**Comment by pgawlowicz** [comment]
Please follow this tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/flexric/source/index.html#id2
The important part is to change the GCC to version 10.

**Reply by kenjo138** [reply]
I followed the tutorial and updated the GCC to version 10. When I run the UE the only thing I am missing is the pdu session completion. I get a message indicating Failed to setup/configure GW interface.
I am still not able to run the xapp. I get the same results as stated above.

**Reply by pgawlowicz** [reply]
Did you install FlexRIC dependencies? see here https://gitlab.eurecom.fr/mosaic5g/flexric

**Reply by kenjo138** [reply]
Yes. I previously had FlexRIC running with OAI. I reinstalled it with the patch.

**Reply by kenjo138** [reply]
I set up ip forwarding for the UE using sudo ip netns add ue1 so now I am getting a PDU session, but the xapp is the only thing left for me to solve.

**Reply by kenjo138** [reply]
All working now. I was running from within the flexric directory. I should have been running from within the flexric/build directory. Problem solved. Thanks for your help.

**Reply by pgawlowicz** [reply]
ok, thanks for the feedback. I will update the tutorial.
